core/src/main/java/org/infinispan/interceptors/distribution/BaseDistributionInterceptor.java

```diff
@@ -243,7 +246,17 @@ public void configure() {
          if (rvrl != null) {
             rvrl.remoteValueNotFound(key);
          }
-         return null;
+         if (!hasSuccesfulResponse) {
+            RpcException rpcException = new RpcException("No successful response: " + responses);
```

Comment:
Indeed. Actually we used to send an `UnsureResponse` back in such cases (with a check in `DistributionResponseGenerator`). But because `ClusteredGetCommand` doesn't have a topology, the check we did there was almost guaranteed to be false, so I finally removed it with ISPN-5463. 

Besides, it's weird to have a locality check in the response generator, I'd rather remove the `ResponseGenerator` and create the `Response` in the command itself. (Or maybe in the interceptors...)